src/interfaces/vault.spec.ts

```diff
@@ -0,0 +1,868 @@
+          lensAdaptersVaultsV2PositionsOfMock.mockRejectedValueOnce(new Error("positionsOf error"));
+          lensAdaptersVaultsV2PositionsOfMock.mockResolvedValue([position]);
+
+          const actualPositionsOf = await vaultInterface.positionsOf("0x001", ["0x001"]);
+
+          expect(actualPositionsOf).toEqual([position]);
+          expect(lensAdaptersVaultsV2PositionsOfMock).toHaveBeenCalledTimes(2);
+          expect(lensAdaptersVaultsV2PositionsOfMock).toHaveBeenCalledWith("0x001", ["0x001"], undefined);
+          expect(vaultInterface.getStatic).not.toHaveBeenCalled();
+        });
+      });
+
+      describe("when the addresses filter was not provided", () => {
+        it("should return all positions and get the static vaults", async () => {
+          lensAdaptersVaultsV2PositionsOfMock.mockRejectedValueOnce(new Error("positionsOf error"));
+          lensAdaptersVaultsV2PositionsOfMock.mockResolvedValue([position]);
+
+          const actualPositionsOf = await vaultInterface.positionsOf("0x001");
+
+          expect(actualPositionsOf).toEqual([position]);
+          expect(lensAdaptersVaultsV2PositionsOfMock).toHaveBeenCalledTimes(2);
+          expect(lensAdaptersVaultsV2PositionsOfMock).toHaveBeenCalledWith("0x001", ["0x001"], undefined);
+          expect(vaultInterface.getStatic).toHaveBeenCalledTimes(1);
+          expect(vaultInterface.getStatic).toHaveBeenCalledWith(undefined, undefined);
+        });
+      });
+    });
+  });
+
+  describe("summaryOf", () => {
+    it("should get the Vaults User Summary for a particular address", async () => {
+      const earningsUserData = createMockEarningsUserData();
+      earningsAccountAssetsDataMock.mockResolvedValueOnce(earningsUserData);
+
+      const actualSummaryOf = await vaultInterface.summaryOf("0x001");
+
+      expect(actualSummaryOf).toEqual({ earnings: "1", estimatedYearlyYield: "1", grossApy: 1, holdings: "1" });
+      expect(earningsAccountAssetsDataMock).toHaveBeenCalledTimes(1);
+      expect(earningsAccountAssetsDataMock).toHaveBeenCalledWith("0x001");
+    });
+  });
+
+  describe("metadataOf", () => {
+    beforeEach(() => {
+      const earningsUserData = createMockEarningsUserData();
+      earningsAccountAssetsDataMock.mockResolvedValueOnce(earningsUserData);
+    });
+
+    describe("when an addresses array is not given", () => {
+      it("should get the Vault User Metadata for a particular address", async () => {
+        const actualMetadataOf = await vaultInterface.metadataOf("0x001");
+
+        expect(actualMetadataOf).toEqual([{ assetAddress: "0x001", earned: "1" }]);
+        expect(earningsAccountAssetsDataMock).toHaveBeenCalledTimes(1);
+        expect(earningsAccountAssetsDataMock).toHaveBeenCalledWith("0x001");
+      });
+    });
+
+    describe("when an addresses array is given", () => {
+      describe("when the address provided is included in the addresses array", () => {
```

Comment:
btw for this cases when a describe is only enclosing one `it` we can just have the `it` and skip the `describe`.

`Describe` are mostly used when you want to set the context for several `it`'s

Reply:
I was trying to avoid longer test names and group them together, otherwise the `it` would look something like 

```typescript
it("should get the Vault User Metadata for the address provided when the address provided is included in the addresses array", async () => {});
```

I'm giving the context, but added just one test case for the time being, could potentially be extended to have more. What would your suggestion be for addressing the long test name without the describe setting the context to be `describe("when an addresses array is given")`

Reply:
So the first describe is actually ok, since it enclosing several `describe` or `its`, the bottom one in particular could be `should get the Vault User Metadata when it is passed a specific address`.

I've got no hard opinions on this, so if you prefer it this way it's ok by me :)

Reply:
I see, makes sense. I've refactored that block with https://github.com/yearn/yearn-sdk/pull/221/commits/0bda5b980f4259f9cc388f83dc2f9bca285a0363

Moving forward I'll try and have less describes with just one test case, I think your reasoning is sensible 👍